presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/DetermineJoinDistributionType.java

```diff
@@ -56,7 +56,7 @@ private static DistributionType determineDistributionType(JoinNode node, Context
             return PARTITIONED;
         }
 
-        if (node.getCriteria().isEmpty() && type == INNER) {
+        if (node.getCriteria().isEmpty() && (type == INNER || type == LEFT)) {
```

Comment:
how do right/full outer joins without criteria get executed?

Reply:
right/full is supported by hash-join (`LookupJoinOperator`) but not by `NestedLoopJoinOperator`, so they are executed as hash-join on constant.
(btw they can optionally have non-equi filter)